src/helpers/helpers.dom.js

```diff
@@ -150,21 +150,41 @@ export function getMaximumSize(canvas, bbWidth, bbHeight, aspectRatio) {
   };
 }
 
+/**
+ * @param {import('../core/core.controller').default} chart
+ * @param {number} [forceRatio]
+ * @param {boolean} [forceStyle]
+ * @returns {boolean} True if redraw is needed.
```

Comment:
nit: the `retinaScale` method doesn't really know about the calling context (i.e. redraw). I would reword this to:

> True if the canvas context size has changed.

Reply:
Used your wording with a small addition:

> True if the canvas context size or transformation has changed